ibis/backends/oracle/__init__.py

```diff
@@ -179,6 +180,18 @@ def _metadata(self, query: str) -> Iterable[tuple[str, dt.DataType]]:
 
         name = util.gen_name("oracle_metadata")
 
+        # If query is a table, adjust the query accordingly
+        sg_expr = sg.parse_one(query, dialect="oracle")
```

Comment:
To make this maybe a bit less confusing, you can tell sqlglot how to parse the query:

```python
try:
    sg_expr = sg.parse_one(query, into=sg.exp.Table, dialect="oracle")
except ParseError:  # or whatever this error should be
    sg_expr = sg.parse_one(query, dialect="oracle")
```

It's a bit repetitive, but I think the intent is a bit more clear ("try to parse this query as a table first")

Reply:
That said, this isn't a blocker!

Reply:
This is a good suggestion and requires less head-scratching.  Updated!